src/NLog/Internal/XmlHelper.cs

```diff
@@ -58,7 +59,33 @@ public static class XmlHelper
         /// </summary>
         private static string RemoveInvalidXmlChars(string text)
         {
-            return String.IsNullOrEmpty(text) ? "" : InvalidXmlChars.Replace(text, "");
+            if (string.IsNullOrEmpty(text))
+                return string.Empty;
+
+#if NET3_5
+            return InvalidXmlChars.Replace(text, "");
+#else
+            StringBuilder sb = null;
+            for (int i = 0; i < text.Length; ++i)
+            {
+                char ch = text[i];
+                if (XmlConvert.IsXmlChar(ch))
+                {
+                    if (sb != null)
+                        sb.Append(ch);
```

Comment:
AFAIK substring is in some cases more efficient then for-append, but maybe not in this case.

Anyway, found a better solution, using `StringBuilder.Append(char[] value, int startIndex, int charCount)`

```c#
            for (int i = 0; i < text.Length; ++i)
            {
                char ch = text[i];
                if (!XmlConvert.IsXmlChar(ch))
                {
                    // Invalid char found - allocated string-builder for cleaning
                    if (sb == null)
                    {
                        //create a fill with already seen chars
                        sb = new StringBuilder(text.Length);
                        sb.Append(text, 0, i);
                    }
                    else
                    {
                        sb.Append(ch);
                    }
                }
            }
```